src/hooks/transaction/types/strategies.ts

```diff
@@ -0,0 +1,22 @@
+import { InterBtcApi } from '@interlay/interbtc-api';
+
+import { Transaction } from '.';
+
+interface StrategiesDepositAction {
+  type: Transaction.STRATEGIES_DEPOSIT;
+  args: Parameters<InterBtcApi['loans']['lend']>;
+}
+
+interface StrategiesWithdrawAction {
+  type: Transaction.STRATEGIES_WITHDRAW;
+  args: Parameters<InterBtcApi['loans']['withdraw']>;
+}
+
+interface StrategiesWithdrawAllAction {
+  type: Transaction.STRATEGIES_ALL_WITHDRAW;
+  args: Parameters<InterBtcApi['loans']['withdrawAll']>;
+}
```

Comment:
This kind of definition won't work long-term. As the next strategy with leverage will have to utilize different extrinsics that are batched. I suggest we should instead define the extrinsic that is used for each `StrategyType`. So we can define it like `Transaction.STRATEGIES.[strategyType].['deposit', 'withdraw', 'withdrawAll']`. Or do you have any other suggestion?

Reply:
I do agree with you and this will change in the leverage position, I just needed to add this because I needed to show a description for the transaction. I will address this in our next PR for the next strat.